spec/models/generic_announcement_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe GenericAnnouncement, type: :model do
+  describe '.default_scope' do
+    before do
+      create_list(:system_announcement, 3)
+      create_list(:instance_announcement, 3)
+    end
+
+    it 'orders by descending valid_from within type' do
+      system, instance = GenericAnnouncement.valid.partition { |a| a.is_a? SystemAnnouncement }
+
+      system_dates = system.map(&:valid_from)
+      sorted_dates = system_dates.sort { |a, b| b <=> a }
+      expect(system_dates).to eq(sorted_dates)
+
+      instance_dates = instance.map(&:valid_from)
+      sorted_dates = instance_dates.sort { |a, b| b <=> a }
+      expect(instance_dates).to eq(sorted_dates)
+    end
+
+    it 'orders system announcements before instance announcements' do
+      announcements = GenericAnnouncement.valid
```

Comment:
This can be more concise.

The rule in Ruby (and most functional languages) is to favour intent over implementation. You should be doing that..

I'm not sure what the exact method is, try googling around?